assets/js/googlesitekit/widgets/datastore/widgets.js

```diff
@@ -95,30 +80,18 @@ export const actions = {
 		invariant( component, 'component is required to register a widget.' );
 		invariant( Object.values( WIDGET_WIDTHS ).includes( width ), `Widget width should be one of: ${ WidgetWidthKeys }, but "${ width }" was provided.` );
 
-		const registry = yield commonActions.getRegistry();
-		let registryKey = yield registry.select( CORE_SITE ).getRegistryKey();
-
-		if ( registryKey === undefined ) {
-			registryKey = uuidv4();
-			yield registry.dispatch( CORE_SITE ).setRegistryKey( registryKey );
-		}
-
-		// We do this assignment in the action rather than the reducer because we can't send a
-		// payload that includes a React component to the reducer; we'll get an error about
-		// payloads needing to be plain objects.
-		if ( WidgetComponents[ registryKey ] === undefined ) {
-			WidgetComponents[ registryKey ] = {};
-		}
-		if ( WidgetComponents[ registryKey ][ slug ] === undefined ) {
-			WidgetComponents[ registryKey ][ slug ] = component;
-		}
-
-		yield {
-			payload: { slug, settings: { priority, width, wrapWidget } },
+		return {
+			payload: {
+				slug,
+				component,
+				settings: {
```

Comment:
Why split these two out, despite `component` being part of `settings` anyway? I think this doesn't make sense anymore now that we don't have the extra handling.